src/api/Buffer.test.ts

```diff
@@ -69,159 +69,188 @@ describe('Buffer API', () => {
+        // One empty line
+        expect(await buffer.length).toEqual(1);
+        expect(await buffer.lines).toEqual(['']);
+      })
+    );
+
+    it(
+      'changes buffer options',
+      withBuffer([], async buffer => {
+        const initial = await buffer.getOption('copyindent');
+        buffer.setOption('copyindent', true);
+        expect(await buffer.getOption('copyindent')).toBe(true);
+        buffer.setOption('copyindent', false);
+        expect(await buffer.getOption('copyindent')).toBe(false);
+
+        // Restore option
+        buffer.setOption('copyindent', initial);
+        expect(await buffer.getOption('copyindent')).toBe(initial);
+      })
+    );
 
-    it('returns null if variable is not found', withBuffer( 
-        [], async buffer => {
-            const test = await buffer.getVar('test');
-            expect(test).toBe(null);
-    }));
+    it(
+      'returns null if variable is not found',
+      withBuffer([], async buffer => {
+        const test = await buffer.getVar('test');
+        expect(test).toBe(null);
+      })
+    );
 
-    it('can set and delete a b: variable to an object', withBuffer(
-        [], async buffer => {
-            buffer.setVar('test', { foo: 'testValue' });
+    it(
+      'can set and delete a b: variable to an object',
+      withBuffer([], async buffer => {
+        buffer.setVar('test', { foo: 'testValue' });
 
-            expect(await buffer.getVar('test')).toEqual({ foo: 'testValue' });
+        expect(await buffer.getVar('test')).toEqual({ foo: 'testValue' });
 
-            expect(await nvim.eval('b:test')).toEqual({ foo: 'testValue' });
-            buffer.deleteVar('test');
+        expect(await nvim.eval('b:test')).toEqual({ foo: 'testValue' });
+        buffer.deleteVar('test');
 
-            expect(await nvim.eval('exists("b:test")')).toBe(0);
+        expect(await nvim.eval('exists("b:test")')).toBe(0);
 
-            expect(await buffer.getVar('test')).toBe(null);
-    }));
+        expect(await buffer.getVar('test')).toBe(null);
+      })
+    );
 
     it('can get list of commands', async () => {
       expect(await nvim.buffer.commands).toEqual({});
     });
 
-
     // TODO: How do we run integration tests for add/clear highlights? and get mark
   });
 
   describe('Chainable API calls', () => {
-
-    it('gets the current buffer name using api chaining', withBuffer(
-        [],
-        async buffer => {
-            await( buffer.name = 'test3.js' );
-            expect(await buffer.name).toMatch('test3.js');
-
-            await buffer.name.then(name => {
-                expect(name).toMatch('test3.js');
-            });
-        }
-    ));
+    it(
+      'gets the current buffer name using api chaining',
+      withBuffer([], async function buffer() {
+        await (buffer._name = 'test3.js');
+        expect(await buffer._name).toMatch('test3.js');
+
+        await (() => {
+          expect(buffer._name).toMatch('test3.js');
```

Comment:
`await` expects an expression, including callbacks. If `nvim.buffer.name` is a promise, then it's okay to use `.then` but I changed `buffer` to a named function. `buffer._name` is now a string and strings don't support Promise API with `.then`.